packages/trip-details/src/index.tsx

```diff
@@ -0,0 +1,247 @@
+import flatten from "flat";
+// @ts-expect-error FIXME: Create TypeScript types for core-utils packages.
+import coreUtils from "@opentripplanner/core-utils";
+import moment from "moment";
+import React, { ReactElement } from "react";
+import { FormattedMessage, FormattedNumber } from "react-intl";
+import { CalendarAlt, Heartbeat, MoneyBillAlt } from "styled-icons/fa-solid";
+
+import * as Styled from "./styled";
+import TripDetail from "./trip-detail";
+import { CaloriesDetailsProps, TripDetailsProps } from "./types";
+
+// Load the default messages.
+import defaultEnglishMessages from "../i18n/en-US.yml";
+
+// HACK: We should flatten the messages loaded above because
+// the YAML loaders behave differently between webpack and our version of jest:
+// - the yaml loader for webpack returns a nested object,
+// - the yaml loader for jest returns messages with flattened ids.
+const defaultMessages: Record<string, string> = flatten(defaultEnglishMessages);
+
+/**
+ * Format text bold (used with FormattedMessage).
+ */
+// TODO: Find a better place for this utility.
+function BoldText(contents: ReactElement): ReactElement {
+  return <b>{contents}</b>;
+}
+
+/**
+ * Helper function to specify the link to dietary table.
+ */
+function dietaryLink(contents: ReactElement): ReactElement {
+  return (
+    <a
+      href="https://health.gov/dietaryguidelines/dga2005/document/html/chapter3.htm#table4"
+      rel="noopener noreferrer"
+      target="_blank"
+    >
+      {contents}
+    </a>
+  );
+}
+
+/**
+ * Default rendering if no component is provided for the CaloriesDetails
+ * slot in the TripDetails component.
+ */
+function DefaultCaloriesDetails({
+  bikeSeconds,
+  calories,
+  walkSeconds
+}: CaloriesDetailsProps): ReactElement {
+  return (
+    <FormattedMessage
+      defaultMessage={defaultMessages["otpUi.TripDetails.caloriesDescription"]}
+      description="Text describing how the calories relate to the walking and biking duration of a trip."
+      id="otpUi.TripDetails.caloriesDescription"
+      values={{
+        b: BoldText,
+        bikeMinutes: Math.round(bikeSeconds / 60),
+        calories: Math.round(calories),
+        dietaryLink,
+        walkMinutes: Math.round(walkSeconds / 60)
+      }}
+    />
+  );
+}
+
+/**
+ * Renders trip details such as departure instructions, fare amount, and calories spent.
+ */
+export function TripDetails({
+  CaloriesDetails = DefaultCaloriesDetails,
+  className = "",
+  DepartureDetails = null,
+  FareDetails = null,
+  itinerary
+}: TripDetailsProps): ReactElement {
+  let companies = "";
+  itinerary.legs.forEach(leg => {
```

Comment:
This may be unrelated to your work, but might be cleaner as a `filter()` and `map()`?

Reply:
Or even a reducer

Reply:
Refactored in https://github.com/opentripplanner/otp-ui/pull/281/commits/d39e81032da5df5266a7dfd0e664940c9cdbdd5f.